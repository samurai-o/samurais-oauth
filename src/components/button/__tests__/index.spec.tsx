import "jest-styled-components";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { Button } from "..";
import { ButtonElement } from "../index.styled";



describe("button 组件测试", () => {
	test("ButtonElement 样式组件快照", () => {
		const dom = render(<ButtonElement type="button" pattern="default" >按钮</ButtonElement>);
		expect(dom.asFragment()).toMatchSnapshot();
	});

	test("Button disabled false测试", () => {
		const click = jest.fn();
		const dom = render(<Button htmlType="button" disabled={false} pattern="default" loading={false} onClick={click}>按钮</Button>);
		fireEvent.click(dom.getByRole("button"));
		expect(click).toBeCalled();
	});

	test("button 快照", () => {
		const click = jest.fn();
		const dom = render(<Button htmlType="button" disabled pattern="default" onClick={click}>测试</Button>);
		expect(dom.asFragment()).toMatchSnapshot();
	});

	test("link 快照", () => {
		const dom = render(<Button htmlType="button" pattern="link" disabled>链接</Button>);
		expect(dom.getByText("链接")).toBeInTheDocument();
		expect(dom.asFragment()).toMatchSnapshot();
	});

	test("link类型组件样式测试", () => {
		const dom = render(<ButtonElement data-testid="ButtonTypeTest" pattern="default" type="button">forbidden</ButtonElement>);
		const styledom = dom.queryByTestId("ButtonTypeTest");
		expect(styledom).toHaveStyleRule("background", "#fff");
		expect(styledom).toHaveStyleRule("box-shadow", "6px 0 16px -8px rgb(0 0 0 / 8%),9px 0 28px 0 rgb(0 0 0 / 0%)");
	});
});