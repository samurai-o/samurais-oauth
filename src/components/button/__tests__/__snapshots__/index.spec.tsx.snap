// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 组件测试 ButtonElement 样式组件快照 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  width: 100%;
  outline: none;
  border: 0px;
  margin: 0px;
  border-radius: 4px;
  padding: 10px 14px;
  background: #fff;
  cursor: pointer;
  color: #484646;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  background: #fff;
  box-shadow: 6px 0 16px -8px rgb(0 0 0 / 8%),9px 0 28px 0 rgb(0 0 0 / 0%);
}

<button
    class="c0"
    pattern="default"
    type="button"
  >
    按钮
  </button>
</DocumentFragment>
`;

exports[`button 组件测试 button 快照 1`] = `
<DocumentFragment>
  .c2 {
  display: block;
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
}

.c2::before {
  display: block;
  content: "";
  width: 0;
  height: 4px;
  background: #000000a6;
  -webkit-animation-name: kjLLRh;
  animation-name: kjLLRh;
  -webkit-animation-duration: 10s;
  animation-duration: 10s;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: backwards;
  animation-fill-mode: backwards;
}

.c0 {
  position: relative;
  width: 100%;
  outline: none;
  border: 0px;
  margin: 0px;
  border-radius: 4px;
  padding: 10px 14px;
  background: #fff;
  cursor: pointer;
  color: #484646;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  background: #fff;
  box-shadow: 6px 0 16px -8px rgb(0 0 0 / 8%),9px 0 28px 0 rgb(0 0 0 / 0%);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
    class="c0"
    pattern="default"
    type="button"
  >
    <div
      class="c1"
    >
      测试
      <span
        class="c2"
      />
    </div>
  </button>
</DocumentFragment>
`;

exports[`button 组件测试 link 快照 1`] = `
<DocumentFragment>
  <a
    class=""
  >
    链接
  </a>
</DocumentFragment>
`;
